pkg/dockerregistry/server/metrics/metrics.go

```diff
@@ -13,26 +13,44 @@ import (
 )
 
 const (
-	registryNamespace = "openshift"
-	registrySubsystem = "registry"
-)
+	namespace = "imageregistry"
 
-var (
-	registryAPIRequests *prometheus.HistogramVec
+	pullthroughSubsystem = "pullthrough"
 )
 
-// Register the metrics.
-func Register() {
-	registryAPIRequests = prometheus.NewHistogramVec(
+var (
+	requestDurationSeconds = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
-			Namespace: registryNamespace,
-			Subsystem: registrySubsystem,
+			Namespace: namespace,
 			Name:      "request_duration_seconds",
-			Help:      "Request latency summary in microseconds for each operation",
+			Help:      "Request latency in seconds for each operation.",
 		},
 		[]string{"operation", "name"},
 	)
-	prometheus.MustRegister(registryAPIRequests)
+	pullthroughRepositoryDurationSeconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: pullthroughSubsystem,
+			Name:      "repository_duration_seconds",
+			Help:      "Latency of operations with remote registries in seconds.",
+		},
+		[]string{"registry", "operation"},
+	)
+	pullthroughRepositoryErrorsTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: pullthroughSubsystem,
+			Name:      "repository_errors_total",
+			Help:      "Cumulative number of failed operations with remote registires.",
+		},
+		[]string{"registry", "operation"},
```

Comment:
Can this contain error type as well (e.g. `BlobUnknown`) or even response's http code?

Reply:
AFAIK errors from the repository client aren't specified, therefore it's not straightforward to categories them.

Reply:
AFAIK the client returns either [errcode.Error](https://github.com/docker/distribution/blob/master/registry/api/errcode/errors.go#L99:6) or [errcode.Errors](https://github.com/docker/distribution/blob/master/registry/api/errcode/errors.go#L176) for API errors (e.g. https://github.com/docker/distribution/blob/master/registry/client/errors.go#L57). They contain http status code and a short message similar to our kstatus.reason (e.g. [`UNAUTHORIZED`](https://github.com/docker/distribution/blob/master/registry/api/errcode/register.go#L39)).

For any other error, "unknown" could be logged.